lib/tests/formulae_dependents.rb

```diff
@@ -169,8 +169,16 @@ def install_dependent(dependent, testable_dependents, args:, build_from_source:
         test "brew", "linkage", "--test", dependent.full_name
 
         if testable_dependents.include? dependent
-          test "brew", "install", "--only-dependencies", "--include-test",
-               dependent.full_name
+          test "brew", "install", "--only-dependencies", "--include-test", dependent.full_name
+
+          dependent.recursive_dependencies.each do |dependency|
```

Comment:
One question (not blocking): are `dependent.recursive_dependencies` tagged with `:build` or `:test` correctly?

Last I checked it may not be: a runtime dependency of a build-only dependency which doesn't also appear as a (possibly recursive) runtime dependency isn't always tagged as build-only.

I might be misremembering though.